app/controllers/routes_controller.rb

```diff
@@ -1,13 +1,55 @@
 class RoutesController < ApplicationController
-  helper_method :routes
+  helper_method :routes, :route
 
   def index
     @model_attributes = model_attributes(Route)
   end
 
+  def new
+    @route = Route.new
+  end
+
+  def create
+    @route = Route.new(route_params)
+
+    if route.save
+      redirect_to routes_path
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    @route = Route.find(params[:id])
+  end
+
+  def update
+    @route = Route.find(params[:id])
+
+    if route.update(route_params)
```

Comment:
```suggestion
    if @route.update(route_params)
```
same as above please